### DISCUSSION TITLE
2.24.0

### DISCUSSION THREAD

**Discussion by Jolg42** [prisma/prisma#7397] [Q&A]
Today, we are excited to share the `2.24.0` stable release 🎉

🌟 **Help us spread the word about Prisma by starring the repo or [tweeting](https://twitter.com/intent/tweet?text=Check%20out%20the%20latest%20@prisma%20release%20v2.24.0%20%F0%9F%9A%80%0D%0A%0D%0Ahttps://github.com/prisma/prisma/releases/tag/2.24.0) about the release.** 🌟 

### MongoDB gets `Json` and `Enum` Support

We just added `Json` and `enum` support to the MongoDB `provider` for Prisma Client. Here's an example with both:

```prisma
datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["mongodb"]
}

model Log {
  id      String @id @default(dbgenerated()) @map("_id")
  message String
  level   Level  @default(Info)
  meta    Json
}

enum Level {
  Info
  Warn
  Error
}
```

You can then use the generated client like this:

```jsx
await prisma.log.create({
  data: {
    level: "info",
    message: "User signed in",
    meta: { user_id: 1 },
  },
})
```

As a reminder, the `mongodb` provider is still in [Early Access](https://www.prisma.io/docs/about/releases#early-access). If you'd like to use MongoDB with Prisma, please fill out this [2-minute Typeform](https://prisma103696.typeform.com/to/FriDuIeM) and we'll get you an invite to our Getting Started Guide and private Slack channel right away!

### New features for the Prisma Data Platform 

The [Prisma Data Platform](https://cloud.prisma.io) (PDP) helps developers collaborate better in projects that are using the open-source tools. One of its main features today is an [online data browser](https://www.prisma.io/blog/prisma-online-data-browser-ejgg5c8p3u4x).

#### View schema

You can now view your project's schema in order to better understand your application or collaborate with your colleagues. The only roles that can view it are: _Admin_ and _Developer_

<details><summary>Expand for a screenshot of the new schema view</summary>
<img src="https://user-images.githubusercontent.com/329419/120306754-da88f580-c2da-11eb-89dc-f8dc2bfade8c.png" alt="View your Schema"  width="450" />
</details>

#### Delete & edit your project

You can now delete your project from your settings.

<details><summary>Expand for a screenshot of the new settings</summary>
<img src="https://user-images.githubusercontent.com/329419/120306724-cfce6080-c2da-11eb-8af2-d1cb7c21635a.png" alt="Delete your Project" width="450"  />
</details>

You can also edit your Project's URL, so you can now correct any mistakes you might have made while creating your project.

#### Static IPs are now supported

If your database is behind a proxy and you need a static IP to whitelist in order to give access to it, you can now get in touch with us by [creating an issue](https://github.com/prisma/studio/issues/new?assignees=&labels=topic%3A+hosted+data+browser&template=hosted-data-browser-bug-report.md&title=) or sending an email at [`martzoukos@prisma.io`](mailto:martzoukos@prisma.io) and we'll enable it for you. 

Please note that while this feature is freely available now, it will be offered as part of a paid plan in the future (towards the end of '21 or beginning of '22).

### Prisma Client

- [Fix distro detection for raspberry pi](https://github.com/prisma/prisma/issues/5552)
- [SQL Server crashes when using autoincrement with createMany](https://github.com/prisma/prisma/issues/6711)
- [PANIC in query-engine\core\src\interpreter\query_interpreters\nested_read.rs:79:561](https://github.com/prisma/prisma/issues/7007)
- [PANIC in query-engine\core\src\query_graph_builder\read\utils.rs:52:85called `Result::unwrap()` on an `Err` value: FieldNotFound { name: "_count", model: "Post" }](https://github.com/prisma/prisma/issues/7091)
- [N-API feature creates memory leak](https://github.com/prisma/prisma/issues/7092)
- [MongoDB: Many to many relation issue](https://github.com/prisma/prisma/issues/7119)
- [Disallow passing a string as a path in Postgres](https://github.com/prisma/prisma/issues/7136)

### Prisma Migrate

- ["Getting is_identity from Resultrow...." crash on Postgres](https://github.com/prisma/prisma/issues/6983)
- [Disable "db" and "migrate" commands for "mongodb" provider until we support it.](https://github.com/prisma/prisma/issues/7012)
- [Support PlanetScale in Migrate](https://github.com/prisma/prisma/issues/7294)

### Prisma Studio

- [Not working in VSCode (docker) devcontainer](https://github.com/prisma/studio/issues/687)
- [Running prisma studio behind a nginx https reverse proxy doesn't work](https://github.com/prisma/studio/issues/688)

### Prisma Engines

- [Add Parameter to the NAPI Constructor to enable LOG_QUERIES](https://github.com/prisma/prisma-engines/issues/1877)
- [Parse type names in datamodel parser at the pest level instead of regular expressions](https://github.com/prisma/prisma-engines/issues/1917)
- [[N-API] Export Functional Engine Commands along side QueryEngine Constructor](https://github.com/prisma/prisma-engines/issues/1926)
- [Make migration ordering stable](https://github.com/prisma/prisma-engines/issues/1942)

## Credits

Huge thanks to @Sytten for helping!

## 📺 Join us for another "What's new in Prisma" livestream

Learn about the latest release and other news from the Prisma community by joining us for another ["What's new in Prisma"](https://www.youtube.com/watch?v=nTI4jjsyoEg&ab_channel=Prisma) livestream.

The stream takes place [on Youtube](https://www.youtube.com/watch?v=nTI4jjsyoEg&ab_channel=Prisma) on **Thursday, March 04** at **5pm Berlin | 8am San Francisco**.

## 🌎 Prisma Day is coming

Save the date for [Prisma Day 2021](https://www.prisma.io/day) and join us for two days of talks and workshops by the most exciting members of the Prisma community. 

- June 29th: Workshops
- June 30th: Talks

We look forward to seeing you there!

<hr /><em>This discussion was created from the release <a href='https://github.com/prisma/prisma/releases/tag/2.24.0'>2.24.0</a>.</em>

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
